src/vcpkg/commands.export-port.cpp

```diff
@@ -0,0 +1,280 @@
+        }
+
+        for (const auto& file : Util::fmap(port_files, [&source](StringView&& str) -> Path {
+                 return str.substr(source.generic_u8string().size() + 1);
+             }))
+        {
+            const auto src_path = source / file;
+            const auto dst_path = destination / file;
+
+            fs.create_directories(dst_path.parent_path(), ec);
+            if (ec)
+            {
+                Checks::msg_exit_with_error(
+                    VCPKG_LINE_INFO, msgExportPortFailedToCreateDirectory, msg::path = dst_path.parent_path());
+            }
+
+            fs.copy_file(src_path, dst_path, CopyOptions::overwrite_existing, ec);
+            if (ec)
+            {
+                Checks::msg_exit_with_error(VCPKG_LINE_INFO,
+                                            msgExportPortFailedToCopyFiles,
+                                            msg::package_name = port_name,
+                                            msg::path = dst_path);
+            }
+        }
+    }
+
+    void export_registry_port(const VcpkgPaths& paths,
+                              StringView port_name,
+                              Optional<Version> version,
+                              const Path& destination)
+    {
+        const auto& config = paths.get_configuration();
+        auto registries = config.instantiate_registry_set(paths);
+        auto maybe_source = version ? registries->fetch_port_files(port_name, version.value_or_exit(VCPKG_LINE_INFO))
+                                    : registries->fetch_port_files(port_name);
+        if (!maybe_source)
+        {
+            msg::println_error(maybe_source.error());
+            Checks::exit_fail(VCPKG_LINE_INFO);
+        }
+
+        auto source = *maybe_source.get();
+        copy_port_files(paths.get_filesystem(), port_name, source.local_portfiles_path, destination);
+        print_success_message(destination, port_name, source.version, source.registry_id);
+        Checks::exit_success(VCPKG_LINE_INFO);
+    }
+
+    void export_classic_mode_versioned(const VcpkgPaths& paths,
+                                       StringView port_name,
+                                       const Version& version,
+                                       const Path& destination)
+    {
+        const auto db_file = paths.builtin_registry_versions / fmt::format("{}-/{}.json", port_name[0], port_name);
+
+        auto& fs = paths.get_filesystem();
+        if (!fs.exists(db_file, VCPKG_LINE_INFO))
+        {
+            msg::println_error(msgExportPortVersionsDbFileMissing, msg::package_name = port_name, msg::path = db_file);
+            Checks::exit_fail(VCPKG_LINE_INFO);
+        }
+
+        auto contents = fs.read_contents(db_file, VCPKG_LINE_INFO);
+        auto maybe_db = parse_git_versions_file(contents, db_file);
+        if (!maybe_db)
+        {
+            msg::println_error(maybe_db.error());
```

Comment:
```suggestion
            msg::println(maybe_db.error());
```

`error()` already has `error:`